mountpoint-s3-crt/src/auth/signing_config.rs

```diff
@@ -34,11 +35,61 @@ impl Debug for SigningConfigInner {
 /// in an Arc so that we can make copies. If there were a way to convert Pin<Box<_>> into Pin<Arc<_>>
 /// then we could avoid needing both.
 #[derive(Debug, Clone)]
-pub struct SigningConfig(pub(crate) Arc<Pin<Box<SigningConfigInner>>>);
+pub struct SigningConfig(Arc<Pin<Box<SigningConfigInner>>>);
 
 impl SigningConfig {
     /// Get out the inner pointer to the signing config
     pub(crate) fn to_inner_ptr(&self) -> *const aws_signing_config_aws {
         &Pin::as_ref(&self.0).get_ref().inner
     }
+
+    /// Builder for [SigningConfig]
+    pub fn builder(region: &str, credentials_provider: CredentialsProvider) -> SigningConfigBuilder {
```

Comment:
We will also need atleast signing name(s3/s3-object-lambda etc) or even scheme name(sigv4/sigv4a).